org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/XMLFormatterDocumentNew.java

```diff
@@ -325,103 +324,114 @@ public void formatChildren(DOMNode currentDOMNode, XMLFormattingConstraints pare
 		}
 	}
 
-	public void formatAttributeValue(DOMAttr attr, XMLFormatterDocumentNew formatterDocument, int indentLevel,
-			XMLFormattingOptions formattingOptions, List<TextEdit> edits) {
+	public void formatAttributeValue(DOMAttr attr, int indentLevel, List<TextEdit> edits) {
 		if (formatterParticipants != null) {
 			for (IFormatterParticipant formatterParticipant : formatterParticipants) {
 				try {
-					if (formatterParticipant.formatAttributeValue(attr, formatterDocument, indentLevel,
-							formattingOptions, edits)) {
+					if (formatterParticipant.formatAttributeValue(attr, this, indentLevel, getFormattingSettings(),
+							edits)) {
 						return;
 					}
 				} catch (Exception e) {
-					LOGGER.log(Level.SEVERE,
-							"Error while processing format attributes for the participant '"
-									+ formatterParticipant.getClass().getName() + "'.", e);
+					LOGGER.log(Level.SEVERE, "Error while processing format attributes for the participant '"
+							+ formatterParticipant.getClass().getName() + "'.", e);
 				}
 			}
 		}
 	}
 
-	void removeLeftSpaces(int to, List<TextEdit> edits) {
-		replaceLeftSpacesWith(to, "", edits);
+	public void removeLeftSpaces(int leftLimit, int to, List<TextEdit> edits) {
+		replaceLeftSpacesWith(leftLimit, to, "", edits);
 	}
 
-	public void removeLeftSpaces(int from, int to, List<TextEdit> edits) {
-		replaceLeftSpacesWith(from, to, "", edits);
-	}
-
-	void replaceLeftSpacesWithOneSpace(int to, List<TextEdit> edits) {
-		replaceLeftSpacesWith(to, " ", edits);
-	}
-
-	public void replaceLeftSpacesWithOneSpace(int from, int to, List<TextEdit> edits) {
-		replaceLeftSpacesWith(from, to, " ", edits);
-	}
-
-	void replaceLeftSpacesWith(int to, String replacement, List<TextEdit> edits) {
-		replaceLeftSpacesWith(-1, to, replacement, edits);
+	public void replaceLeftSpacesWithOneSpace(int leftLimit, int to, List<TextEdit> edits) {
+		replaceLeftSpacesWith(leftLimit, to, " ", edits);
 	}
 
 	void replaceLeftSpacesWith(int leftLimit, int to, String replacement, List<TextEdit> edits) {
 		int from = getLeftWhitespacesOffset(leftLimit, to);
-		createTextEditIfNeeded(from, to, replacement, edits);
+		if (from >= 0) {
+			createTextEditIfNeeded(from, to, replacement, edits);
+		}
 	}
 
-	void replaceQuoteWithPreferred(int from, int to, String replacement, List<TextEdit> edits) {
-		createTextEditIfNeeded(from, to, replacement, edits);
+	void replaceQuoteWithPreferred(int from, int to, List<TextEdit> edits) {
+		createTextEditIfNeeded(from, to, getQuotationAsString(), edits);
 	}
 
+	/**
+	 * Returns the offset of the first whitespace is found from the given range
+	 * [leftLimit,to].
+	 * 
+	 * When :
+	 * <ul>
+	 * <li>leftLimit is equals to -1 (ex : an element tag name which have not a

```

Comment:
... the leftLimit is set **to** 'to - 1'

Reply:
I remove this comment which is now outdated.